tools/audio/audio_tablegen.c

```diff
@@ -290,19 +304,330 @@ tablegen_soundfonts(const char *sf_hdr_out, char **soundfonts_paths, int num_sou
     return EXIT_SUCCESS;
 }
 
+/* Sequences */
+
+struct seq_order {
+    size_t num_sequences;
+    char **names;
+    char **enum_names;
+    bool *ptr_flags;
+    void *buf_ptr;
+};
+
+static void
+read_seq_order(struct seq_order *order, const char *path)
+{
+    // Read from file, we assume the file has been preprocessed with cpp so that whitespace is collapsed and each line
+    // has the form:
+    // (name,enum_name) or *(name,enum_name)
+    UNUSED size_t data_size;
+    char *filedata = util_read_whole_file(path, &data_size);
+
+    // We expect one entry per line, gather the total length
+    size_t total_size = 0;
+    for (char *p = filedata; *p != '\0'; p++) {
+        if (*p == '\n') {
+            total_size++;
+        } else if (isspace(*p)) {
+            goto malformed;
+        }
+    }
+
+    // Make required allocations
+    char **name_list      = malloc(total_size * (sizeof(char *) + sizeof(char *) + sizeof(bool)));
+    char **enum_name_list = &name_list[total_size];
+    bool *ptr_flags       = (bool *)&enum_name_list[total_size];
```

Comment:
I don't think merging these into one allocation is worth it, keeping them separate would be easier to read and easier to modify (but maybe this could be an array of structs instead?)